.circleci/config/base.yml

```diff
@@ -1428,48 +1414,55 @@ commands:
                 condition: << parameters.use-out-cache >>
                 steps:
                   - *step-restore-out-cache
-            - *step-gn-gen-default
-            - *step-electron-build
-            - *step-maybe-electron-dist-strip
-            - step-electron-dist-build:
-                additional-targets: shell_browser_ui_unittests third_party/electron_node:headers third_party/electron_node:overlapped-checker electron:hunspell_dictionaries_zip
-
-            - *step-show-goma-stats
-
-            # mksnapshot
-            - *step-mksnapshot-build
-            - *step-maybe-cross-arch-snapshot
-
-            # chromedriver
-            - *step-electron-chromedriver-build
-
-            - when:
-                condition: << parameters.build-nonproprietary-ffmpeg >>
-                steps:
-                  # ffmpeg
-                  - *step-ffmpeg-gn-gen
-                  - *step-ffmpeg-build
-
-      # Save all data needed for a further tests run.
-      - when:
-          condition: << parameters.persist >>
-          steps:
-            - *step-minimize-workspace-size-from-checkout
-            - run: |
-                rm -rf src/third_party/electron_node/deps/openssl
-                rm -rf src/third_party/electron_node/deps/v8
-            - *step-persist-data-for-tests
 
       - when:
           condition: << parameters.build >>
           steps:
-            - *step-maybe-generate-breakpad-symbols
-            - *step-maybe-zip-symbols
-
-      - when:
-          condition: << parameters.build >>
-          steps:
-            - move_and_store_all_artifacts
+            - build_and_save_artifacts:
+                artifact-key: << parameters.artifact-key >>
+                build-nonproprietary-ffmpeg: << parameters.build-nonproprietary-ffmpeg >>
+            - when:
+                condition:
+                  not:
+                    equal: [ 'nil', << parameters.secondary-artifact-key >> ]
+                steps:
+                  - run:
+                      name: Configuring secondary artifact set
+                      command: |
+                        echo 'export GN_EXTRA_ARGS="<< parameters.secondary-artifact-gn-args >> $GN_EXTRA_ARGS"' >> $BASH_ENV
+                        echo 'export MAS_BUILD="true"' >> $BASH_ENV
```

Comment:
if "secondary artifact" really means "MAS artifact" maybe we should name it that? Or we should refactor s.t. `MAS_BUILD=true` isn't needed or is more generic.

Reply:
It doesn't? Specifically this was built with the intention we could re-use it to build two similar artifacts in one build on other jobs too.  E.g. linux-x64 and linux-x64-no-run-as-node

This `MAS_BUILD` thing is a nasty hack but will have no impact on non-macos builds anyway 🤷

Reply:
yeah, my point was let's get rid of the MAS_BUILD hack :)

Reply:
Removed 👍